### ISSUE TITLE
Magic workaround for exceeding 1.5m immersion boundary

### ISSUE THREAD

**Issue by panthurrr** [alvr-org/alvr-visionos#34]
Might be impossible to accomplish but keep an eye out for anything in Apple's documentation to reset the immersion central location if we are reaching its edge.  Perhaps utilizing a combination of ARKit with immersion or if anyone has a Unity Pro sub check if they have found a way to do this.

ALVR server could store the location data temporarily and transpose the view to the new center if we can figure a way to reset location quickly.

**Comment by Fox-McCloud** [comment]
I spoke with Millenomi about this, and it's hard coded; there might be a way to reset it, but I suspect this will cause further issues.

As an example; I tried resetting while in VRChat while utilizing full body tracking; as I suspected, recentering completely destroys playspace alignment. If you have continuous calibration, it'd quickly reset, but this is unideal for most users, and I suspect even those who don't utilize trackers would also have issues.

At the very least it means that controllers would also become unaligned.

**Comment by zarik5** [comment]
> ALVR server could store the location data temporarily and transpose the view to the new center if we can figure a way to reset location quickly.

Probably it's better to keep track on the client itself

**Comment by panthurrr** [comment]
The major issue with resetting now is the ALVR client will lose its connection to the server.  If we can decouple that connection from the immersive space render loop then we are free to dismiss and reopen the immersive space.

**Comment by Kestre333** [comment]
I am not a developer, but Synth Riders in the immersive space allows you to walk around physically. It will briefly show you surroundings but then pop you back into full immersion. I _think_ it has to do with setting it as mixed immersion. Hope that helps.

**Comment by Fox-McCloud** [comment]
> I am not a developer, but Synth Riders in the immersive space allows you to walk around physically. It will briefly show you surroundings but then pop you back into full immersion. I _think_ it has to do with setting it as mixed immersion. Hope that helps.

This is correct; the restrictions on immersive environments are relaxed if you can see "some" of your surroundings.

If Apple doesn't provide us a way to customize their boundary system, we might have to get creative with mixed mode.
